### PR TITLE
Migrates CI process to GitHub actions

### PR DESCRIPTION
**Pull request by SHession**
## What does this change?

This PR is part of the work to migrate our CI process from TeamCity to GithHb Actions, continuing the work begun here: https://github.com/guardian/ophan/pull/5454 (contains relevant links to useful resources). 

## How to test

Does the GitHub action complete successfully? Does the lambda run correctly once deployed?

## How can we measure success?

Another TeamCity build migrated to GitHub actions.

## Have we considered potential risks?

Issues with deploy resulting in our GeoIP files not being updated correctly.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      # Checkout the branch
+      - uses: actions/checkout@v3
+      # Java is needed for the Scala Play app
+      - uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'corretto'
+          cache: 'sbt'
+      # Build CDK and Play (in sequence)
+      - run: sbt clean compile test assembly
```
Comment (deedeeh) [inline .github/workflows/ci.yml:24]: I was looking at the other [geoip-db-refresher](https://github.com/guardian/ophan-geoip-db-refresher/pull/256) PR and it looks different in this step which you changed in the `ci.sh` file but it doesn't have the word `test`. Why is it different?

Comment (SHession) [reply .github/workflows/ci.yml:24]: Good question, simple answer, [geoip-db-refresher](https://github.com/guardian/ophan-geoip-db-refresher) has no tests!

**Review by deedeeh — COMMENTED** [review]
Why the `riff-raff.yaml` is not removed? 

Also how do you know `Your usual build steps here...`? Is it something that I can know from TeamCity or in `build.sbt`? It would be nice to have a chat about that after standup 😅

**Comment by SHession** [comment]
> Why the `riff-raff.yaml` is not removed?
> 
> Also how do you know `Your usual build steps here...`? Is it something that I can know from TeamCity or in `build.sbt`? It would be nice to have a chat about that after standup 😅

I have removed the riff-raff file, good spot!

In terms of the build steps, TeamCity is your best place to check: 
image

**Review by deedeeh — APPROVED** [review]
It looks good 🚀 Thank you Sam for answering my questions in a call.

**Pull request merged**